src/assets/scripts/client/base/Compass.js

```diff
@@ -0,0 +1,57 @@
+import { radians_normalize } from '../math/circle';
+
+/**
+ * @class Compass
+ */
+class Compass {
+    /**
+     * @constructor
+     */
+    // istanbul ignore next
+    constructor() {
+        /**
+         * Magnetic declination, in radians east
+         *
+         * @property _magneticNorth
+         * @type {number}
+         * @default 0
+         */
+        this._magneticNorth = 0;
+    }
+
+    /**
+     * Facade to access the compass's magnetic declination value
+     *
+     * @for Compass
+     * @property magneticNorth
+     * @return {number}
+     */
+    get magneticNorth() {
+        return this._magneticNorth;
+    }
+
+    /**
+     * Fascade to change the compass's magnetic declination value
+     *
+     * @for Compass
+     * @property magneticNorth
+     * @param {number}
+     */
+    set magneticNorth(magneticNorth) {
+        this._magneticNorth = magneticNorth;
+    }
+
+    /**
+     *
+     * @for Compass
+     * @method normalize
+     * @param radians {number}
+     * @return {number} the angle relative to the magnetic north within [0,2π]
+     */
+    normalize(radians) {
```

Comment:
so is this like `.getMagneticHeading(trueHeading)`? If so, please add a description at the top of the docblock to clearly explain what this does. It's pretty apparent from the math, but not from the name (in my opinion).